python/pdo/contract/contract.py

```diff
@@ -67,10 +68,9 @@ def read_from_file(cls, ledger_config, basename, data_dir = None) :
             raise Exception('invalid contract file; {}'.format(filename))
 
         try :
-            state = ContractState.read_from_cache(contract_id, current_state_hash, data_dir=data_dir)
+            state = ContractState.read_from_cache(contract_id, current_state_hash)
```

Comment:
It seems a bit odd to have a read_from_cache but no write_to_cache/save_to_cache.

Reply:
Its always saved to cache.
This is not a block but a contract.